prometheus/summary.go

```diff
@@ -304,10 +309,19 @@ func (s *summary) Observe(v float64) {
 	if len(s.hotBuf) == cap(s.hotBuf) {
 		s.asyncFlush(now)
 	}
+	if s.createdTs == nil {
+		s.setCreatedTimestamp()
+	}
+}
+
+func (s *summary) setCreatedTimestamp() {
```

Comment:
Do we really need this method that is used only once? 

It's a little bit too shallow to me (perhaps adds more complexity not saves complexity).

See: https://thanos.io/tip/contributing/coding-style-guide.md/#keep-the-interface-narrow-avoid-shallow-functions

Reply:
Yeah good point